### ISSUE TITLE
build errors running npm run build-aot

### ISSUE THREAD

**Issue by tomwayson** [beginor/ng2-esri-demo#2]
When I run `npm run build-aot` it works until this line, which fails w/ the following error:
```shell
> ngc --project tsconfig-aot.json

Error at /Users/tom/code/ng2-esri-demo/src/app/services/map.service.ts:3:1: Import assignment cannot be used when targeting ECMAScript 2015 modules. Consider using 'import * as ns from "mod"', 'import {a} from "mod"', 'import d from "mod"', or another module format instead.
Error at /Users/tom/code/ng2-esri-demo/src/app/components/home/home.component.ts:3:1: Import assignment cannot be used when targeting ECMAScript 2015 modules. Consider using 'import * as ns from "mod"', 'import {a} from "mod"', 'import d from "mod"', or another module format instead.
Error at /Users/tom/code/ng2-esri-demo/src/app/components/map3d/map3d.component.ts:3:1: Import assignment cannot be used when targeting ECMAScript 2015 modules. Consider using 'import * as ns from "mod"', 'import {a} from "mod"', 'import d from "mod"', or another module format instead.
Error at /Users/tom/code/ng2-esri-demo/src/app/components/home/home.component.ngfactory.ts:102:18: Property 'mapElement' is private and only accessible within class 'HomeComponent'.
Error at /Users/tom/code/ng2-esri-demo/src/app/components/map3d/map3d.component.ngfactory.ts:102:18: Property 'mapElement' is private and only accessible within class 'Map3dComponent'.
```

**Comment by beginor** [comment]
It is imposible to build this demo with AOT, since I do not know to build dojo/esri with rollup.

**Comment by tomwayson** [comment]
I'm not sure that it's impossible, but you'd have to change a few things.

First, I can get past the above error by changing the syntax for how you require the ArcGIS modules from `import Map = require('esri/Map');` to `import * as Map from 'esri/Map';`. However, I then get a TS error:
```shell
Error at /Users/tom6947/code/ng2-esri-demo/src/app/components/map-view/map-view.component.ngfactory.ts:102:18: Property 'mapElement' is private and only accessible within class 'MapViewComponent'.
Error at /Users/tom6947/code/ng2-esri-demo/src/app/components/scene-view/scene-view.component.ngfactory.ts:102:18: Property 'mapElement' is private and only accessible within class 'SceneViewComponent'.
```

I've never worked w/ `ngc` so I can't be much help there.

If you were able to get `ngc` to run, then the rollup bundling _might_ actually work b/c rollup _should_ just ignore the modules that it can't resolve (esri/dojo). However, you'd have to change how the output bundle is loaded on the page to use a require statement like `require('dist/build.js')`. See my [example project](https://github.com/tomwayson/esri-rollup-example#how-does-it-work).

In other words, I don't think rollup is the problem, or at least, we haven't gotten to that part of the build yet.

**Comment by beginor** [comment]
I tried something with aot build:

- all imports like `import Map = require('esri/map')` changed to `import * as Map from 'esri/Map'`;
- try config rollup to ignore `esri` with plugin `rollup-plugin-ignore`;
- try config rollup to resolve `esri` with plugin `rollup-plugin-bower-resolve`;

my aot build step is:

- [x] npm run tsc
- [x] npm run ngc
- [x] npm run tsc-aot
- [ ] npm run rollup

but still failed at the last rollup step  `rollup -c rollup.config.js`, the error is:

```
Could not resolve 'esri/views/MapView' from /Users/zhang/Projects/ng2-esri-demo/src/app/components/map-view/map-view.com
ponent.js
```

See my last check in for details, do you have any ideas?
